### PR TITLE
Remove fuse_with_relu argument from batch_norm constructor

### LINKED ISSUE #21003: batch_norm, fuse_with_relu, 与act参数互斥，不能用户同时使用
### 用户问题
- 请教一下，paddle中的fluid.layers.batch_norm(),函数 中act这个参数有用吗？设置为 act=‘relu’，那么意味着bn层后面接了relu吗
  - 答：是的
- 追问：是不是一定要act设置完之后，在设置这个fuse_with_relu 。。如果不设置fuse_with_relu，那么只设置act，是不是就不起作用。。
  - 答：我理解这个地方如果使用应该是把relu fuse进去了，而直接用act是另外接了了一个relu op
- 追问：意思是 同时使用这两个参数，act=relu,fuse_with_relu=True..那么就是后面接了两次 relu吗？？
  - 答：刚咨询了下相关同学，这个fuse只在预测用，训练的话还是用act，这个接口需要改进下
- 追问：我这有一个疑问，我看nn.py中 batch_norm（）源码，里面act设置与否，后面都没有使用，感觉是一个多余的参数。
  - 答：解释了下代码
- 追问：那预测的时候 是不是就要设置 act为None，fuse_with_relu=True ？？
  - 答：如果不考虑 mkldnn 的话，只使用act这个参数就好了，训练和预测一样

### 总结
- batch_norm中`fuse_with_relu`只用于mkldnn的预测，是否不应该暴露给用户？ @bingyanghuang  @jianhang-liu #11434 加了这个参数。

### PR DESCRIPTION
**Pull request by wojtuss**
This patch removes the 'fuse_with_relu' argument from the BatchNorm operator constructor. The 'fuse_with_relu' attribute should not be exposed to a user as an argument of the constructor.

fix #21003

### CODE REVIEW

File: python/paddle/fluid/dygraph/nn.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ class Conv2D(layers.Layer):
 
     Returns:
         None
-    
+
```
Comment (luotao1) [inline python/paddle/fluid/dygraph/nn.py]: Please do not change the other lines.

Comment (wojtuss) [reply python/paddle/fluid/dygraph/nn.py]: Done.

**Review by luotao1 — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
```
[14:27:19]	[Step 1/1] API Difference is: 
[14:27:19]	[Step 1/1] - paddle.fluid.layers.batch_norm (ArgSpec(args=['input', 'act', 'is_test', 'momentum', 'epsilon', 'param_attr', 'bias_attr', 'data_layout', 'in_place', 'name', 'moving_mean_name', 'moving_variance_name', 'do_model_average_for_mean_and_var', 'fuse_with_relu', 'use_global_stats'], varargs=None, keywords=None, defaults=(None, False, 0.9, 1e-05, None, None, 'NCHW', False, None, None, None, True, False, False)), ('document', 'dc68d19bdd820eb4f3f00fa460f53203'))
[14:27:19]	[Step 1/1] ?                                                                                                                                                                                                                                             ------------------                                                                                                                                                -------                  ^^  ^  -------------------------
[14:27:19]	[Step 1/1] 
[14:27:19]	[Step 1/1] + paddle.fluid.layers.batch_norm (ArgSpec(args=['input', 'act', 'is_test', 'momentum', 'epsilon', 'param_attr', 'bias_attr', 'data_layout', 'in_place', 'name', 'moving_mean_name', 'moving_variance_name', 'do_model_average_for_mean_and_var', 'use_global_stats'], varargs=None, keywords=None, defaults=(None, False, 0.9, 1e-05, None, None, 'NCHW', False, None, None, None, True, False)), ('document', 'aa78c56e260724058f0e64207300bb19'))
[14:27:19]	[Step 1/1] ?                                                                                                                                                                                                                                                                                                                                                                                                               ^^^^^^ +++++++++ ^^^^^^^^^^^^^
[14:27:19]	[Step 1/1] 
[14:27:19]	[Step 1/1] - paddle.fluid.dygraph.BatchNorm ('paddle.fluid.dygraph.nn.BatchNorm', ('document', 'fb93cdf32f21a1c26e784d61a80051a7'))
[14:27:19]	[Step 1/1] ?                                                                                    ^^^ ^^     ^ ^^^^  ^^ ----------
[14:27:19]	[Step 1/1] 
[14:27:19]	[Step 1/1] + paddle.fluid.dygraph.BatchNorm ('paddle.fluid.dygraph.nn.BatchNorm', ('document', '276037086f3296fa62fe5faae716277d'))
[14:27:19]	[Step 1/1] ?                                                                                    ^^^^ ^^^^   ++ ++ ^^^^ ^  ^^^^^
[14:27:19]	[Step 1/1] 
[14:27:19]	[Step 1/1] - paddle.fluid.dygraph.BatchNorm.__init__ (ArgSpec(args=['self', 'name_scope', 'num_channels', 'act', 'is_test', 'momentum', 'epsilon', 'param_attr', 'bias_attr', 'dtype', 'data_layout', 'in_place', 'moving_mean_name', 'moving_variance_name', 'do_model_average_for_mean_and_var', 'fuse_with_relu', 'use_global_stats', 'trainable_statistics'], varargs=None, keywords=None, defaults=(None, False, 0.9, 1e-05, None, None, 'float32', 'NCHW', False, None, None, True, False, False, False)), ('document', '6adf97f83acf6453d4a6a4b1070f3754'))
[14:27:19]	[Step 1/1] ?                                                                                                                                                                                                                                                                                        ------------------                                                                                                                                                                 -------
[14:27:19]	[Step 1/1] 
[14:27:19]	[Step 1/1] + paddle.fluid.dygraph.BatchNorm.__init__ (ArgSpec(args=['self', 'name_scope', 'num_channels', 'act', 'is_test', 'momentum', 'epsilon', 'param_attr', 'bias_attr', 'dtype', 'data_layout', 'in_place', 'moving_mean_name', 'moving_variance_name', 'do_model_average_for_mean_and_var', 'use_global_stats', 'trainable_statistics'], varargs=None, keywords=None, defaults=(None, False, 0.9, 1e-05, None, None, 'float32', 'NCHW', False, None, None, True, False, False)), ('document', '6adf97f83acf6453d4a6a4b1070f3754'))
[14:27:26]	[Step 1/1] Process exited with code 1
[14:27:26]	[Step 1/1] Process exited with code 1 (Step: Build and test (Command Line))
[14:27:26]	[Step 1/1] Step Build and test (Command Line) failed
```
@lanxianghit Please review the API difference.

**Review by lanxianghit — APPROVED** [review]
LGTM

**Review by heavengate — APPROVED** [review]

**Comment by luotao1** [comment]
1. Please cherry-pick this PR to release/1.6
2. Please update the Chinese batch_norm API document in `FluidDoc` repo，and cherry-pick into release/1.6 in `FluidDoc` repo.   @bingyanghuang
   - https://github.com/PaddlePaddle/FluidDoc/blob/develop/doc/fluid/api_cn/dygraph_cn/BatchNorm_cn.rst
   -  https://github.com/PaddlePaddle/FluidDoc/blob/develop/doc/fluid/api_cn/layers_cn/batch_norm_cn.rst

**Comment by bingyanghuang** [comment]
> 1. Please cherry-pick this PR to release/1.6
> 2. Please update the Chinese batch_norm API document in `FluidDoc` repo，and cherry-pick into release/1.6 in `FluidDoc` repo.   @bingyanghuang
>    
>    * https://github.com/PaddlePaddle/FluidDoc/blob/develop/doc/fluid/api_cn/dygraph_cn/BatchNorm_cn.rst
>    * https://github.com/PaddlePaddle/FluidDoc/blob/develop/doc/fluid/api_cn/layers_cn/batch_norm_cn.rst

1. Done
2. Done https://github.com/PaddlePaddle/FluidDoc/pull/1588
